### PR TITLE
Add Lint with flake8 to CI on ert3 branch

### LINKED ISSUE #1201: Run pylint as part of CI on ert3 module
I've received two comments in #1198 on unused imports and variables. This is food for pylint, not a human reviewer. Let us get `pylint` (or a similar tool like `flake8`) up and running as part of our CI on the `ert3` module.

### PR DESCRIPTION
**Pull request by autt**
**Issue**
Resolves #1201 

**Approach**
A basic setup of lint with flake8 with github actions.

### CODE REVIEW

**Comment by markusdregi** [comment]
@autt: The failing tests are due to: https://github.com/equinor/ert/issues/1215

Until resolved, just rerun them 👍

**Comment by autt** [comment]
> @autt: The failing tests are due to: #1215
> 
> Until resolved, just rerun them +1

Thanks @markusdregi! I became aware when I stared working on #1221, and added (an attempt of) a quick fix there. But i guess it was a better idea making a separate issue :+1:

**Comment by markusdregi** [comment]
A quick fix to improve the chances of success is also fine :) I think we should keep the issue to fixing it "a bit more properly" though...

Seems like the linting is printing a bunch of warnings, but not failing 🤔 Is that intentional?

File: .github/workflows/style.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 ert3 --count --select=E9,F63,F7,F82 --show-source --statistics
```
Comment (markusdregi) [inline .github/workflows/style.yml:34]: Should we remove the select to run validation with all possible errors instead? :) I get this running locally with no errors at least:

```
flake8 ert3 --max-line-length=88 --ignore=F401,E711
```

88 as this is the line length enforced by black... Afterwards, both of the two ignored errors should be pretty straight forward to fix such that we can run it with no ignores.

Comment (autt) [reply .github/workflows/style.yml]: Thanks for your suggestion, I agree that its better to run it like this :) What was there was really just from the Python workflow template ( https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions) that I used as a starting point.

Comment (markusdregi) [reply .github/workflows/style.yml]: I figured as much ;)

File: .github/workflows/style.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest
```
Comment (markusdregi) [inline .github/workflows/style.yml:26]: I guess we don't need pytest?

File: .github/workflows/style.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (markusdregi) [inline .github/workflows/style.yml:27]: Should we just remove the `if`-statement such that the installation fails if the `requirements.txt` disappears?

**Review by markusdregi — APPROVED** [review]
This looks good 👍 Had two minor comments for which you do as you like with... 🎉

**Pull request merged**